### PR TITLE
Redis, Run email notifications via Bull queue, Bull UI

### PR DESCRIPTION
**Pull request by lionsdwarf**
1) Redis config
2) Email notification rollups fetched via Bull cron
3) Notification rollups sent individually via Bull "send" queue
4) Bull-UI job monitor at `/manage/jobs` endpoint
5) Authentication for `/manage/*` routes (required moving server-side middleware functions into `lib/auth/server`)

### CODE REVIEW

File: apps/api/worker.js [hunk 1]
```diff
@@ -41,21 +38,6 @@ const config = [
     cronExp: EVERY_HOUR,
     fn: etl.syncClinikoUsers
   },
-  {
-    cronExp: EVERY_MINUTE,
-    fn: notifications.appointmentConfirmation
-  },
-  {
-    cronExp:
-      process.env.IS_HOURLY_REMINDERS === 'true'
```
Comment (lionsdwarf) [inline apps/api/worker.js]: Bonus: IS_HOURLY_REMINDERS flags are now removed

File: lib/queue.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import Bull from 'bull'
+
+const { REDIS_URL } = process.env
+
+const bull = (name, url = REDIS_URL, opts) => {
```
Comment (craigspaeth) [inline lib/queue.js]: Nicely done wrapping this!

File: lib/queue.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  if (!url) {
+    throw new Error(
```
Comment (craigspaeth) [inline lib/queue.js]: Does the bull lib not throw an error is `url` is missing? Seems like this might be doing dual duty.

File: lib/queue.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      'REDIS_URL not found: Redis connection requires a REDIS_URL.'
+    )
+  }
+  return new Bull(name, url, opts)
+}
+
+export const UIQueues = [
```
Comment (craigspaeth) [inline lib/queue.js]: 👍 nice

File: lib/auth/server.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  }&returnTo=${process.env.APP_URL}`
+  res.redirect(logoutUrl)
+}
+
+export const isAdmin = (tok, usr) =>
+  Boolean(tok && usr && usr.roles && usr.roles.includes('admin'))
+
+export const ensureAdmin = async (req, res, next) => {
+  const auth0Token = getAuth0Token(req)
+  const user = await userFromAuth0Token(auth0Token)
```
Comment (craigspaeth) [inline lib/auth/server.js]: 👏 very nice! Well done extracting this lib and the context lib to re-use the auth verification here.

**Comment by craigspaeth** [comment]
👏 Well done! The only blocking comment is making sure the [script actually runs](https://github.com/findOctave/cortex/pull/1292#discussion_r340297294) by not using the unix `&&`.

**Comment by craigspaeth** [comment]
@lionsdwarf Feel free to merge after addressing the script update.

File: lib/queue.js (hunk 3 above)
Comment (lionsdwarf) [reply lib/queue.js]: No it defaults to the Redis defaults

File: Procfile [hunk 6]
```diff
@@ -1,2 +1,2 @@
 web: npm run web
-worker: npm run worker
\ No newline at end of file
+worker: npm run worker && npm run queue
```
[the remark this answers is not in the document]
Comment (lionsdwarf) [reply Procfile]: 🤦‍♂ 

Good catch. Will fix.

**Pull request merged**